### PR TITLE
dashboard skeleton with codemirror

### PR DESCRIPTION
**Pull request by erizocosmico**
Just review last commit, it depends on #1

### CODE REVIEW

File: src/setupTests.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// CodeMirror needs all of this in order to work.
+// see: https://discuss.codemirror.net/t/working-in-jsdom-or-node-js-natively/138/5
+global.document.body.createTextRange = function() {
+  return {
+    setEnd: function () {},
```
Comment (Serabe) [inline src/setupTests.js]: `setEnd() {}` (and the rest)

**Comment by Serabe** [comment]
Needs topbar.

**Comment by erizocosmico** [comment]
@Serabe topbar is here https://github.com/bblfsh/dashboard/pull/2/files#diff-6f7a94b64ddda80f88c2971ba5b8b1a3R8

**Comment by erizocosmico** [comment]
@Serabe amended with the changes you requested

**Comment by erizocosmico** [comment]
Added resizable panels. Only the topbar missing for fulfilling the specs.

**Comment by erizocosmico** [comment]
Topbar (with tests) implemented. It currently has no design, since all we have is the mockup. Once everything is done we'll have to revisit everything for styling.

Right now, it looks like this:

screenshot_2017-06-21_15 32 59

File: src/App.js [hunk 2]
```diff
@@ -21,24 +21,74 @@ const Content = styled.div`
...
+    const { languages, selectedLanguage, code, ast, loading, actualLanguage } = this.state;
 
     return (
       <Wrap>
-        <Header />
+        <Header
+          languages={languages}
+          selectedLanguage={selectedLanguage}
+          actualLanguage={actualLanguage}
+          onLanguageChanged={e => this.onLanguageChanged(e)}
```
Comment (dpordomingo) [inline src/App.js]: why not only `onLanguageChanged={this.onLanguageChanged}`

File: src/components/Header.js [hunk 3]
```diff
@@ -1,14 +1,134 @@
...
+`
+
+export function DriverCode({ languages, selectedLanguage, actualLanguage }) {
+  const driver = selectedLanguage === 'auto' 
+    ? actualLanguage
+    : selectedLanguage;
+
+  return (
+    <DriverCodeBox>
+      <img src={githubIcon} alt='driver code' />
```
Comment (dpordomingo) [inline src/components/Header.js]: the `alt` should be something describing the image itself, or empty if its not important for the content itself
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img

File: src/components/Header.js [hunk 4]
```diff
@@ -1,14 +1,134 @@
...
+      return (
+        <option value={k} key={k}>{name}</option>
+      );
+    });
+
   return (
     <Container>
-      Babelfish Dashboard
+      <Title>
+        <TitleImage src={bblfshLogo} alt='bblfsh' />
```
Comment (dpordomingo) [inline src/components/Header.js]: alt='bblfsh logo image'

File: src/App.js (hunk 2 above)
Comment (erizocosmico) [inline src/App.js]: Because it's not bound to `this` :(

Comment (dpordomingo) [inline src/App.js]: explained IRL and agree; deleting comment

Comment (dpordomingo) [inline src/App.js]: thnks

**Comment by erizocosmico** [comment]
@dpordomingo amended with your changes

**Comment by Serabe** [comment]
Quick note, we won't have any design.

File: src/components/Header.test.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        onRunParser={onRunParserSpy}
+        actualLanguage='java'
+        selectedLanguage='auto' />
+    );
+
+    wrapper.find('#run-parser').simulate('click');
+    expect(onRunParserSpy.mock.calls.length).toBe(1);
+  });
+
+  it('has the button disabled if is loading', () => {
```
Comment (Serabe) [inline src/components/Header.test.js]: This is not what this code is testing.

Comment (erizocosmico) [inline src/components/Header.test.js]: Yes, it is. It's a snapshot of when `loading` is `true` has the `disabled={true}`

**Comment by erizocosmico** [comment]
@Serabe yep, I know, that's why I think it should be the last point after we're finished.

**Comment by dpordomingo** [comment]
so... can we merge @Serabe ?

File: src/components/Header.test.js (hunk 5 above)
Comment (Serabe) [inline src/components/Header.test.js]: No, it is not. The title suggests that the check should be something like:

```js
expect(component.button.matches(':disabled'))
```

If the markup changes we would need to go to the snapshot to check if the attribute is still the same, every single time, while using the `matches` selector is closer to what you are describing in the title and nobody needs to go looking for the snapshot to check the disabled attribute is still true.

**Comment by erizocosmico** [comment]
@Serabe Again, yes, it is. You don't need to go check anything in the snapshot. When a snapshot changes jest reports the change in a diff way and you can either accept the change directly in the CLI or go change your code to make it pass as it was before. That's how snapshot testing works.

File: src/components/Header.test.js (hunk 5 above)
Comment (erizocosmico) [inline src/components/Header.test.js]: Again, yes, it is. You don't need to go check anything in the snapshot. When a snapshot changes jest reports the change in a diff way and you can either accept the change directly in the CLI or go change your code to make it pass as it was before. That's how snapshot testing works.

Comment (Serabe) [inline src/components/Header.test.js]: I don't care that much.

**Pull request merged**